packages/website/blog/2025-12-31-revamping-the-ban-types-rule.mdx

```diff
@@ -0,0 +1,155 @@
+TypeScript defines uppercase "wrapper" object types corresponding to each of JavaScript's primitive data types and equivalent wrapper classes: `boolean`/`Boolean`, `number`/`Number`, and so on.
+
+```ts
+let myNumber: Number;
+
+myNumber = 0; // Ok
+myNumber = Number(0); // Ok
+
+let myObject: Object = 'allowed by TypeScript'; // Ok
+```
+
+Due to the quirks of TypeScript's structural typing, the corresponding primitives are also assignable to these uppercase types, since they have the same "shape".
+Instances of the uppercase class types also have surprisingly different behaviors at runtime, including always being truthy (even for values like `new Boolean(false)`) and being compared by reference (`new String("a") !== new String("a")`).
+
+Instead of using uppercase wrapper object types, code should generally stick with lower-case primitive types.
+Common cases and more examples are noted in the new [`@typescript-eslint/no-wrapper-object-types`](/rules/no-wrapper-object-types) rule's documentation.
+
+## User-Defined Type Bans
+
+[`@typescript-eslint/no-restricted-types`](/rules/no-restricted-types) is the new rule for banning a configurable list of type names.
+It acts similarly to the old `ban-types` rule:
+
+- It defines no types by default: i.e. it must be configured to be useful
+- It allows specifying types to ban, along with optional messages to display
+
+As with `@typescript-eslint/ban-types`, `@typescript-eslint/no-restricted-types` additionally allows specifying auto-fixes to replace types with.
+The new rule also allows specifying suggestions to surface in editors, for cases when the fix isn't known to be reliable.
+
+Here's an example of configuring the new rule to ban a deprecated API, with a custom message and two suggestions:
+
+```ts
+{
+  "@typescript-eslint/no-restricted-types": [
+    "error",
+    {
+      "types": {
+        "DeprecatedOldAPI": {
+          "message": "Use either NewAPIOne or NewAPITwo instead",
+          "suggest": ["NewAPIOne", "NewAPITwo"],
+        },
+      },
+    },
+  ],
+}
+```
+
+By acting as a purely user-configured rule, `@typescript-eslint/no-restricted-types`'s behavior falls in line with core ESLint rules like [`no-restricted-imports`](https://eslint.org/docs/latest/no-restricted-imports) and [`no-restricted-syntax`](https://eslint.org/docs/latest/no-restricted-syntax).
+
+See the new [`@typescript-eslint/no-restricted-types`](/rules/no-restricted-types) rule's documentation for more information on how to configure it.
+
+## Closing Thoughts
+
+We'd like to give a big thanks to all the users who gave feedback on the old `@typescript-eslint/ban-types` rule.
+In particular, we'd like to thank [Ryan Cavanaugh](https://github.com/RyanCavanaugh) from the TypeScript team for advising on the rule's design (and for dealing with users frequently sending issues to TypeScript itself on what was actually an overly aggressive lint rule).
```

Comment:
I need to check with Ryan (tagging @RyanCavanaugh) about this. On the one hand, Ryan (and presumably other TS folks) have dealt with a lot of inbound because of `ban-type` rules. On the other hand, I don't know how well `no-empty-object-type` (especially still being in the recommended preset) addresses the TS team's feedback. And even if it mostly-to-completely does, is it OK to mention a TS team lead here - does that misleadingly imply approval from the TS team?

I'll ask on Monday, after the holiday weekend is over.